diff --git a/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc b/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc
index b3a35db..74e87b9 100644
--- a/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc
+++ b/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc
@@ -95,8 +95,12 @@ do_provide_mender_defines() {
             ;;
     esac
 
-    # Strip leading and trailing whitespace, then newline divide, and remove dtbo's.
-    MENDER_DTB_NAME="$(echo "${KERNEL_DEVICETREE}" | sed -r 's/(^\s*)|(\s*$)//g; s/\s+/\n/g' | sed -ne '/\.dtbo$/b; p')"
+    if [ -n "${MENDER_MENDER_DTB_NAME_FORCE}" ]; then
+        MENDER_DTB_NAME="${MENDER_MENDER_DTB_NAME_FORCE}"
+    else
+	    # Strip leading and trailing whitespace, then newline divide, and remove dtbo's.
+	    MENDER_DTB_NAME="$(echo "${KERNEL_DEVICETREE}" | sed -r 's/(^\s*)|(\s*$)//g; s/\s+/\n/g' | sed -ne '/\.dtbo$/b; p')"
+    fi
 
     if [ -z "$MENDER_DTB_NAME" ]; then
         bbfatal "Did not find a dtb specified in KERNEL_DEVICETREE"
